src/plus/gk/account/organizationService.ts

```diff
@@ -0,0 +1,120 @@
+	}
+
+	dispose(): void {
+		this._disposable.dispose();
+	}
+
+	@gate()
+	async getOrganizations(options?: {
+		force?: boolean;
+		accessToken?: string;
+		userId?: string;
+	}): Promise<Organization[] | null | undefined> {
+		const scope = getLogScope();
+		const userId = options?.userId ?? (await this.container.subscription.getSubscription(true))?.account?.id;
+		if (userId == null) {
+			this.updateOrganizations(undefined);
+			return this._organizations;
+		}
+
+		if (this._organizations === undefined || options?.force) {
+			if (!options?.force) {
+				this.loadStoredOrganizations(userId);
+				if (this._organizations != null) return this._organizations;
+			}
+
+			let rsp;
+			try {
+				rsp = await this.connection.fetchApi(
+					'user/organizations-light',
+					{
+						method: 'GET',
+					},
+					{ token: options?.accessToken },
+				);
+			} catch (ex) {
+				debugger;
+				Logger.error(ex, scope);
+
+				void window.showErrorMessage(`Unable to get organizations due to error: ${ex}`, 'OK');
+				this.updateOrganizations(undefined);
+				return this._organizations;
+			}
+
+			if (!rsp.ok) {
+				debugger;
+				Logger.error('', scope, `Unable to get organizations; status=(${rsp.status}): ${rsp.statusText}`);
+
+				void window.showErrorMessage(`Unable to get organizations; Status: ${rsp.statusText}`, 'OK');
+
+				// Setting to null prevents hitting the API again until you reload
+				this.updateOrganizations(null);
+				return this._organizations;
+			}
+
+			const organizationsResponse = (await rsp.json()) as OrganizationsResponse;
+			const organizations = organizationsResponse.map((o: any) => ({
+				id: o.id,
+				name: o.name,
+				role: o.role,
+			}));
+
+			await this.storeOrganizations(organizations, userId);
+			this.updateOrganizations(organizations);
+		}
+
+		return this._organizations;
+	}
+
+	@gate()
+	private loadStoredOrganizations(userId: string): void {
+		const storedOrganizations = this.container.storage.get(`gk:${userId}:organizations`);
+		if (storedOrganizations == null) return;
+		const { timestamp, data: organizations } = storedOrganizations;
+		if (timestamp == null || Date.now() - timestamp > organizationsCacheExpiration) {
+			return;
+		}
+
+		this.updateOrganizations(organizations);
+	}
+
+	private async storeOrganizations(organizations: Organization[], userId: string): Promise<void> {
+		return this.container.storage.store(`gk:${userId}:organizations`, {
+			v: 1,
+			timestamp: Date.now(),
+			data: organizations,
+		});
+	}
+
+	private onSubscriptionChanged(e: SubscriptionChangeEvent): void {
+		if (e.current?.account?.id == null) {
+			this.updateOrganizations(undefined);
+		}
+	}
+
+	private updateOrganizations(organizations: Organization[] | null | undefined): void {
+		this._organizations = organizations;
+		void setContext('gitlens:gk:hasMultipleOrganizationOptions', (organizations ?? []).length > 1);
```

Comment:
Why "Options"?

Reply:
That's from back when you had an extra 'default' option that we decided to remove. I can update to `hasOrganizations`.

Reply:
Done in commit: https://github.com/gitkraken/vscode-gitlens/commit/651c5adf49adcdbeab5080fb52a14937a5d5c3a2